src/Core/Manifest/ModuleManifest.php

```diff
@@ -225,4 +238,31 @@ public function getModules()
     {
         return $this->modules;
     }
+
+    /**
+     * Get modules sorted by priority
+     *
+     * @return Module[]
+     */
+    public function getSortedModules()
```

Comment:
I suggest to do the sorting in ModuleManfiest::regenerate(); That way you can cache the sorting as well (yay performance).